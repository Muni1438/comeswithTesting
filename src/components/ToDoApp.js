import React,{useReducer,useState} from 'react'
import "./ToDoApp.scss"
export const Actions={
    ADD_TODO:"ADD_TODO",
    DELETE_TODO:"DELETE_TODO"
}
const initialState=[]
export const reducer=(state,action)=>{
switch(action.type){
    case Actions.ADD_TODO:
        return [...state,addingData(action.payload)]
    case Actions.DELETE_TODO:
        return state.filter(mine =>mine.id !== action.payload.id)
    default:
        return state
}
}
const addingData=(Text)=>{
    return{
        id:Math.random(),
        Text
    }
}
function ToDoApp() {
    const [Text, setText] = useState(initialState)
    const [state, dispatch] = useReducer(reducer, initialState)
    const adding=()=>{
        return{
            type:Actions.ADD_TODO,
            payload:Text
        }
    }
    const submitHandular=(e)=>{
        e.preventDefault();
        setText("")
        dispatch(adding())
    }
    const deleting=(dataID)=>{
        return{
            type:Actions.DELETE_TODO,
            payload:{id:dataID}
        }
    }
    return (
        <div>
        <form data-testid="form" onSubmit={submitHandular}>
        <h1>Add Todo Here</h1>
            <input placeholder="Add Todo To Display" className="inputTag" type="text" value={Text} onChange={e=>setText(e.target.value)} />
        </form>
        {state.length?(state.map(data=><h2 data-testid="Text" className="h1Tag" key={data.id} onClick={()=>dispatch(deleting(data.id))}>{data.Text}</h2>)):<h3>Nothing To Delete</h3>}
        </div>
    )
}

export default ToDoApp
