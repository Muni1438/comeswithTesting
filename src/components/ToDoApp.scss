.h1Tag{
    border: 2px solid rgb(47, 136, 209);
    background-color: rgb(218, 99, 208);
    width: fit-content;
    margin: 10px auto 10px auto;
    color: rgb(46, 34, 32);
}
.inputTag{
    text-decoration: none;
    outline-color: violet;
    border-top: none;
    border-left: none;
    border-right: none;
    border-bottom: 1px solid teal;
}