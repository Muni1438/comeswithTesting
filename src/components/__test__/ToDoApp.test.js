import React from 'react'
import {render,cleanup} from "@testing-library/react"
import ToDoApp from '../ToDoApp'
import ReactDom from "react-dom"

afterEach(cleanup);
test("renders Correctly",()=>{
    const dammal= document.createElement("div");
    ReactDom.render(<ToDoApp/>,dammal)
    expect(dammal.querySelector("h1").textContent).toBe("Add Todo Here");
})

test("my todo snapshot ",()=>{
    const {asFragment}= render(<ToDoApp/>)

    expect(asFragment(<ToDoApp/>)).toMatchSnapshot()
})