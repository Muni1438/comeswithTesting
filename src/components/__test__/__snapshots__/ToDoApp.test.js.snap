// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my todo snapshot  1`] = `
<DocumentFragment>
  <div>
    <form>
      <h1>
        Add Todo Here
      </h1>
      <input
        class="inputTag"
        placeholder="Add Todo  Display Here"
        type="text"
        value=""
      />
    </form>
  </div>
</DocumentFragment>
`;
